pkg/cloud/vsphere/provisioner/govmomi/create.go

```diff
@@ -292,8 +309,13 @@ func (pv *Provisioner) cloneVirtualMachine(s *SessionContext, cluster *clusterv1
 	// all.
 	diskMap := func(diskSpecs []vsphereconfigv1.DiskSpec) map[string]int64 {
 		diskMap := make(map[string]int64)
-		for _, s := range diskSpecs {
-			diskMap[s.DiskLabel] = s.DiskSizeGB
+		for i, s := range diskSpecs {
+			diskLabel := s.DiskLabel
+			if len(diskLabel) == 0 {
```

Comment:
great!

Reply:
Was one of the things initially tripping me up when first deploying. 👍